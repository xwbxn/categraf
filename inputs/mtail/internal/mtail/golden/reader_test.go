package golden

import (
	"os"
	"sync"
	"testing"
	"time"

	"flashcat.cloud/categraf/inputs/mtail/internal/metrics"
	"flashcat.cloud/categraf/inputs/mtail/internal/metrics/datum"
	"flashcat.cloud/categraf/inputs/mtail/internal/testutil"
)

var expectedMetrics = metrics.MetricSlice{
	{
		Name:    "bytes_total",
		Program: "reader_test",
		Kind:    metrics.Counter,
		Keys:    []string{"operation"},
		LabelValues: []*metrics.LabelValue{
			{
				Labels: []string{"sent"},
				Value:  datum.MakeInt(62793673, time.Date(2011, 2, 23, 5, 54, 10, 0, time.UTC)),
			},
			{
				Labels: []string{"received"},
				Value:  datum.MakeInt(975017, time.Date(2011, 2, 23, 5, 54, 10, 0, time.UTC)),
			},
		},
	},
	{
		Name:    "connections_total",
		Program: "reader_test",
		Kind:    metrics.Counter,
		Keys:    []string{},
		LabelValues: []*metrics.LabelValue{
			{
				Value: datum.MakeInt(52, time.Date(2011, 2, 22, 21, 54, 13, 0, time.UTC)),
			},
		},
	},
	{
		Name:    "connection-time_total",
		Program: "reader_test",
		Kind:    metrics.Counter,
		Keys:    []string{},
		LabelValues: []*metrics.LabelValue{
			{
				Value: datum.MakeInt(1181011, time.Date(2011, 2, 23, 5, 54, 10, 0, time.UTC)),
			},
		},
	},
	{
		Name:    "transfers_total",
		Program: "reader_test",
		Kind:    metrics.Counter,
		Keys:    []string{"operation", "module"},
		LabelValues: []*metrics.LabelValue{
			{
				Labels: []string{"send", "module"},
				Value:  datum.MakeInt(2, time.Date(2011, 2, 23, 5, 50, 32, 0, time.UTC)),
			},
			{
				Labels: []string{"send", "repo"},
				Value:  datum.MakeInt(25, time.Date(2011, 2, 23, 5, 51, 14, 0, time.UTC)),
			},
		},
	},
	{
		Name:        "foo",
		Program:     "reader_test",
		Kind:        metrics.Gauge,
		Keys:        []string{"label"},
		LabelValues: []*metrics.LabelValue{},
	},
	{
		Name:    "bar",
		Program: "reader_test",
		Kind:    metrics.Counter,
		Keys:    []string{},
		LabelValues: []*metrics.LabelValue{
			{
				Value: datum.MakeInt(0, time.Unix(0, 0)),
			},
		},
	},
	{
		Name:    "floaty",
		Program: "reader_test",
		Kind:    metrics.Gauge,
		Type:    metrics.Float,
		Keys:    []string{},
		LabelValues: []*metrics.LabelValue{
			{
				Labels: []string{},
				Value:  datum.MakeFloat(37.1, time.Date(2017, 6, 15, 18, 9, 37, 0, time.UTC)),
			},
		},
	},
	{
		Name:    "stringy",
		Program: "reader_test",
		Kind:    metrics.Text,
		Type:    metrics.String,
		Keys:    []string{},
		LabelValues: []*metrics.LabelValue{
			{
				Labels: []string{},
				Value:  datum.MakeString("hi", time.Date(2018, 6, 16, 18, 4, 0, 0, time.UTC)),
			},
		},
	},
}

func TestReadTestData(t *testing.T) {
	f, err := os.Open("reader_test.golden")
	testutil.FatalIfErr(t, err)
	defer f.Close()
	readMetrics := ReadTestData(f, "reader_test")
	testutil.ExpectNoDiff(t, expectedMetrics, readMetrics, testutil.SortSlices(metrics.Less), testutil.IgnoreUnexported(metrics.Metric{}, sync.RWMutex{}, datum.String{}))
}
