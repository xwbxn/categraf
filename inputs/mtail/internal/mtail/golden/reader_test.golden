counter bytes_total {operation=sent} 62793673 2011-02-23T05:54:10Z
counter bytes_total {operation=received} 975017 2011-02-23T05:54:10Z
counter connections_total 52 2011-02-22T21:54:13Z
counter connection-time_total 1181011 2011-02-23T05:54:10Z
counter transfers_total {operation=send,module=module} 2 2011-02-23T05:50:32Z
counter transfers_total {operation=send,module=repo} 25 2011-02-23T05:51:14Z
gauge foo {label=}
counter bar
gauge floaty 37.1 2017-06-15T18:09:37Z
text stringy hi 2018-06-16T18:04:00Z
